include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         * The values and derivatives are computed for every B-splines at the given coordinate x.
+         * @param[out] derivs The values and n derivatives of the B-splines evaluated at coordinate x. It has to be a (1+degree)*(1+n) 2D mdspan.
+         * @param[in] x The coordinates where B-splines derivatives are evaluated.
+         * @param[in] n The number of derivatives to evaluate (in addition to the B-splines values themselves).
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type eval_basis_and_n_derivs(
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a (1+nbasis) 1D mdspan.
+         * @return The values of the integrals.
+         */
         template <class Layout, class MemorySpace2>
         KOKKOS_INLINE_FUNCTION ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2>
         integrals(
                 ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2> int_vals) const;
 
+        /** @brief Returns the coordinate of the knot corresponding to the given index for a B-spline.
+         *
+         * @param[in] knot_idx Integer identifying index of the knot.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
         }
 
+        /** @brief Returns the coordinate of the first support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_first_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid()));
         }
 
+        /** @brief Returns the coordinate of the last support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_last_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + degree() + 1));
         }
 
+        /** @brief Returns the coordinate of the \f$n\f$-th knot associated to a DiscreteElement identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @param[in] n Integer indexing a knot in the support of the B-spline.
+         * @return Coordinate of the knot.
+         */
```

Comment:
```suggestion
        /** @brief Returns the coordinate of the first knot in the support of the identified B-spline.
         *
         * Each B-spline has a support defined over (degree+2) knots. For a B-spline identified by the
         * provided DiscreteElement, this function returns the (n+1)-th knot in the support of the B-spline.
         *
         * @param[in] ix DiscreteElement identifying the B-spline.
         * @param[in] n Index of the relevant  knot from the support of the B-spline.
         * @return Coordinate of the knot.
         */
```
Not sure about the description of `n`. Feel free to change if you have a better suggestion